tools/Dashboard/DevHome.Dashboard/Views/AddWidgetDialog.xaml.cs

```diff
@@ -225,14 +223,29 @@ private void SelectFirstWidgetByDefault()
         if (selectedTag as WidgetDefinition is WidgetDefinition selectedWidgetDefinition)
         {
             _selectedWidget = selectedWidgetDefinition;
-            await ViewModel.SetWidgetDefinition(selectedWidgetDefinition, ActualTheme);
+            await ViewModel.SetWidgetDefinition(selectedWidgetDefinition);
         }
         else if (selectedTag as WidgetProviderDefinition is not null)
         {
             ViewModel.Clear();
         }
     }
 
+    [RelayCommand]
+    internal async Task UpdateThemeAsync()
+    {
+        // Update the icons for each available widget listed.
+        foreach (var providerItem in AddWidgetNavigationView.MenuItems.Cast<NavigationViewItem>())
+        {
+            foreach (var widgetItem in providerItem.MenuItems.Cast<NavigationViewItem>())
+            {
+                var widgetDefinition = widgetItem.Tag as WidgetDefinition;

```

Comment:
Do we need to try catch widget definition usage, or we don't expect RPC errors here?

Reply:
Technically yes, I think we should, but I think the only way to do this without making the whole dashboard a mess of try/catches is to wrap the widget objects in a model so we don't have to be querying the real objects the entire time (which I do plan to do at some point).